Frameworks/Foundation/NSIndexPath.mm

```diff
@@ -215,54 +223,45 @@ - (NSInteger)section {
 }
 
 /**
- @Status Stub
-*/
-- (NSIndexPath*)indexPathByRemovingLastIndex {
-    UNIMPLEMENTED();
-    return self;
-}
-
-/**
- @Status Stub
- @Notes
-*/
-- (instancetype)init {
-    UNIMPLEMENTED();
-    return StubReturn();
-}
-
-/**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (void)getIndexes:(NSUInteger*)indexes range:(NSRange)positionRange {
-    UNIMPLEMENTED();
+    if (positionRange.location + positionRange.length > [self length]) {
+        [NSException raise:NSRangeException
+                    format:@"-[%s %s]: range {%lu, %lu} beyond bounds (%lu)",
+                           class_getName([self class]),
+                           sel_getName(_cmd),
+                           positionRange.location,
+                           positionRange.length,
+                           [self length]];
+    }
+
+    std::copy(_indexes.cbegin() + positionRange.location, _indexes.cbegin() + positionRange.location + positionRange.length, indexes);
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 + (BOOL)supportsSecureCoding {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return YES;
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (id)initWithCoder:(NSCoder*)decoder {
-    UNIMPLEMENTED();
-    return StubReturn();
+    NSData* data = [decoder decodeObjectOfClass:[NSData class] forKey:s_archiveKey];
+    return [self initWithIndexes:static_cast<const NSUInteger*>([data bytes]) length:([data length] / sizeof(NSUInteger))];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (void)encodeWithCoder:(NSCoder*)coder {
-    UNIMPLEMENTED();
+    NSData* dataToEncode = [self length] > 0L ?
```

Comment:
We can't do this; especially if we want peoples' preferences to eventually roam to ARM and other platforms whose NSIntegers are different from x86.